src/psygnal/_signal.py

```diff
@@ -1105,10 +1106,21 @@ def resume(self, reducer: ReducerFunc | None = None, initial: Any = _NULL) -> No
         if not getattr(self, "_args_queue", None):
             return
         if reducer is not None:
-            if initial is _NULL:
-                args = reduce(reducer, self._args_queue)
+            if (
+                hasattr(reducer, "__annotations__")
+                and len(reducer.__annotations__) == 2
+            ):
+                args = cast(Callable[[List[tuple]], tuple], reducer)(self._args_queue)
```

Comment:
this is my only concern at the moment.  I can break tests, for example, just by removing the type annotation from the reduce_events

```python
def _reduce_events(li: list[tuple]): ...  # works
def _reduce_events(li): ...  # doesn't work
```

that's a bit too flimsy of a contract